src/apps/lwaftr/icmp.lua

```diff
@@ -121,5 +115,6 @@ function new_icmpv6_packet(from_eth, to_eth, from_ip, to_ip, initial_pkt, l2_siz
    local ip_pl_p = new_pkt.data + l2_size + constants.o_ipv6_payload_len
    wr16(ip_pl_p, C.ntohs(new_ipv6_len))
 
+   ipv6_header:free()
```

Comment:
Nit: I think it's possible to free a header as soon as it has been pushed into a datagram, like in the write_icmp4 case. Is there this linebv had to be moved below? If not I prefer it's like in the write_icmp4 case.